### ISSUE TITLE
Stack [amplify-XXX] already exists when amplify init 

### ISSUE THREAD

**Issue by nelmasry** [aws-amplify/amplify-cli#14017]
### How did you install the Amplify CLI?

npm (npm install -g @aws-amplify/cli@latest)

### If applicable, what version of Node.js are you using?

18.x

### Amplify CLI Version

12.13.0 & 10.0.0

### What operating system are you using?

ubuntu-latest

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

Nothing, suddenly it stopped working.

### Describe the bug

I'm working with amplify with github actions to automate deployments and was working fine, suddenly it started to fail showing me error that `Stack [amplify-XXX] already exists when amplify init `
I have checked https://github.com/aws-amplify/amplify-cli/issues/11120#issuecomment-1270038231 and tried downgrading amplify cli to 10.0.0 as mentioned in the thread but it didn't fix the issue, it just changed the error to 
`AlreadyExistsException: Stack [amplify-xxx] already exists
    at Request.extractError (/snapshot/repo/build/node_modules/aws-sdk/lib/protocol/query.js:50:29)
    at Request.callListeners (/snapshot/repo/build/node_modules/aws-sdk/lib/sequential_executor.js:106:20)`

### Expected behavior

Pipeline works normally and deploy to existing environments without issues.

### Reproduction steps

Create github actions pipeline to deploy amplify existing env

### Log output

<details>

```
# Put your logs below this line


```

</details>

### Before submitting, please confirm:

- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

**Comment by ykethan** [comment]
Hey @nelmasry, thank you for reaching out. Could you try using `amplify pull --appId <app-id> --envName next --yes` instead of `amplify init --appId <app-id> --envName next --yes` and let us know if this mitigates the issue.

**Comment by nelmasry** [comment]
hi @ykethan , yes i have tried but for some reason amplify pull command can not find the app with the sent app Id, although i run amplify list apps and app exists, even when send same aws credentials and region
![image](https://github.com/user-attachments/assets/31a5bf9b-0efd-4c46-9561-d0a2450d2422)

**Comment by ykethan** [comment]
Hey @nelmasry, could you try adding a `--debug` to the pull command. Additionally, using the same aws credential if you run `amplify pull` without an appId or envName does amplify provide a list of amplify apps?

**Comment by nelmasry** [comment]
@ykethan 
When i tried pull without app Id i got this error message
`Current environment cannot be determined.
Resolution: Use 'amplify init' in the root of your app directory to create a new environment.
Learn more at: https://docs.amplify.aws/cli/project/troubleshooting/`

**Comment by ykethan** [comment]
@nelmasry apologies outside your existing project directory. The `amplify pull` should provide you prompt to select from a list apps present in the account. could you test it in your local environment with similar credentials and region to verify if it does provide the app-id.
Additionally, is the workflow currently using AWS SSO credentials?

**Comment by nelmasry** [comment]
hi @ykethan 
After investigating the issues I encountered, I discovered they were all related to an invalid `team-provider-info.json` file:

- The error `Cannot read properties of undefined (reading 'Region')` occurred because one of the environments was missing the AWS Region property (for some reason).
- For the current issue, [Stack [amplify-XXX] already exists](https://github.com/aws-amplify/amplify-cli/issues/14017), it was also caused by a missing environment in the `team-provider-info.json` file. Once I added the environment, everything worked fine without needing to use amplify pull or other workarounds.

Honestly, I'm not sure why you didn’t suggest checking this file, as it seems to be the root cause of such issues. This was also the case in other bug reports like [this one](https://github.com/aws-amplify/amplify-cli/issues/11120).

Anyway, thank you for your support and efforts to help. The issue is now resolved.
